### PR TITLE
added first solution... or test to solution... nvm

### PR DESCRIPTION
**Pull request by IhorRome**

### CODE REVIEW

File: src/main/java/core/basesyntax/Calculator.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    private static final char DIVIDING = '/';
+    private static final char POWER = '^';
+
+    public double calculate(double firstNumber, double secondNumber, char operator) {
+        double result;
+        switch (operator) {
+            case ADDITION:
+                result = firstNumber + secondNumber;
+                if (result >= Double.MAX_VALUE || result <= -Double.MAX_VALUE) {
+                    throw new ArithmeticException();
```
Comment (andriiromaniuk) [inline src/main/java/core/basesyntax/Calculator.java:19]: ArithmeticException is exception which is throwed when we try to do something that math itself forbids, this is not the case.

File: src/main/java/core/basesyntax/exception/NullDivisorException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.exception;
+
+public class NullDivisorException extends RuntimeException {
```
Comment (andriiromaniuk) [inline src/main/java/core/basesyntax/exception/NullDivisorException.java:3]: just use arithmetic, this is the case.

File: src/main/java/core/basesyntax/exception/InvalidOperatorException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.exception;
+
+public class InvalidOperatorException extends RuntimeException {
```
Comment (andriiromaniuk) [inline src/main/java/core/basesyntax/exception/InvalidOperatorException.java:3]: UnsupportedOperationException also exists

**Review by andriiromaniuk — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        assertEquals(expected, actual);
+        actual = calculator.calculate(20, 10, '-');
+        expected = 10;
+        assertEquals(expected, actual);
+        actual = calculator.calculate(10, 20, '*');
+        expected = 200;
+        assertEquals(expected, actual);
+        actual = calculator.calculate(10, 20, '/');
+        expected = 0.5;
+        assertEquals(expected, actual);
```
Comment (andriiromaniuk) [inline src/test/java/core/basesyntax/CalculatorTest.java:34]: You are testing half of your code with one Unit(!) test, make it different tests.

**Review by andriiromaniuk — CHANGES_REQUESTED** [review]

**Review by andriiromaniuk — APPROVED** [review]

File: src/main/java/core/basesyntax/Calculator.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    private static final char MULTIPLYING = '*';
+    private static final char DIVIDING = '/';
+    private static final char POWER = '^';
+
+    public double calculate(double firstNumber, double secondNumber, char operator) {
+        double result;
+        switch (operator) {
+            case ADDITION:
+                result = firstNumber + secondNumber;
+                if (result >= Double.MAX_VALUE || result <= -Double.MAX_VALUE) {
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/Calculator.java:19]: this is overthinking. besides, there is no such thing as ValueLimit, you'll get positive or negative infinity. we don't need this check

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Calculator.java (hunk 5 above)
Comment (Max-Pochepets) [reply src/main/java/core/basesyntax/Calculator.java:19]: what about this?

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

**Review by Max-Pochepets — APPROVED** [review]

**Pull request closed without merging**